spring-cloud-deployer-spi/src/main/java/org/springframework/cloud/deployer/spi/core/AppDeploymentRequest.java

```diff
@@ -92,12 +94,8 @@ public AppDeploymentRequest(AppDefinition definition, Resource resource,
 		Assert.notNull(resource, "resource must not be null");
 		this.definition = definition;
 		this.resource = resource;
-		this.deploymentProperties = deploymentProperties == null
-				? Collections.<String, String>emptyMap()
-				: Collections.unmodifiableMap(deploymentProperties);
-		this.commandlineArguments = commandlineArguments == null
-				? Collections.<String>emptyList()
-				: Collections.unmodifiableList(commandlineArguments);
+		this.deploymentProperties = deploymentProperties;
```

Comment:
Isn't it still be possible to have `null` values for `deploymentProperties` and `commandlineArguments` when this constructor is invoked `directly` with `null` values?